### ISSUE TITLE
[HOLD for payment 2024-08-20][$500] WS Chat – Blank page opens when go back from IOU thread in a Collect WS chat

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#40937]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.65-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+jp_e_category@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Log in as Collect WS Employee
3. In a WS chat create a Submit expense
4. Open just created Submit expense
5. Navigate back to WS chat via header link

## Expected Result:
Collect WS chat with history opens

## Actual Result:
Blank page opens when go back from IOU thread in a Collect WS chat

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/ca52fdf6-bcf5-400d-be4a-ff4a86d269a2

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01dee105cffad79099</li>
        <li>Upwork Job ID: 1783269604755496960</li>
        <li>Last Price Increase: 2024-06-11</li>
<li>Automatic offers: </li>
<ul>
<li>jjcoffee | Contributor | 0</li>
<li>tsa321 | Contributor | 103410989</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @thesahindia</details>

**Comment by lanitochka17** [comment]
@sakluger  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by beodw** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
WS Chat – Blank page opens when go back from IOU thread in a Collect WS chat

### What is the root cause of that problem?
The reportsActions array is temporarily empty in the ReportActionsView while loading new report data on slow connections.

### What changes do you think we should make in order to solve the problem?
In the reportsActionsView component we can return a skeleton loader instead of null when the reportActions array is temporarily empty.

In ReportsActionsView.tsx:

``` if (!reportActions.length) {
        return null;
    }
  ```
  Changes to:
  
  ``` if (!reportActions.length) {
        return <ReportActionsSkeletonView />;
    }
  ```

### What alternative solutions did you explore? (Optional)
None

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by beodw** [comment]
@jjcoffee @sakluger Hi guys, kindly following up on whether this issue has been resolved. Please, kindly let me know when you will have the capacity to review it.

**Comment by jjcoffee** [comment]
@beodw Thanks for the [proposal](https://github.com/Expensify/App/issues/40937#issuecomment-2077724149)! 

I think the expected result here is probably not to show a loader, as we actually already have all the data we need. It looks like there's something specifically wrong with loading via the route that highlights the expense's `reportAction`, i.e. `/r/{reportID}/{reportActionID}`. If you omit the latter part, it loads immediately as expected (regardless of connection speed).

https://github.com/Expensify/App/assets/27287420/aa089042-5465-48e7-8a4a-0c8f3e52de91

**Comment by sakluger** [comment]
@beodw @charles-liang do you have any updated proposals based on @jjcoffee's comment?

**Comment by jjcoffee** [comment]
@charles-liang Apologies for not giving feedback on your [proposal](https://github.com/Expensify/App/issues/40937#issuecomment-2076298113). We don't need to implement caching as such. You can see from the issue's video that tapping on the report directly loads it just fine (as I also point out [here](https://github.com/Expensify/App/issues/40937#issuecomment-2082249005)).

**Comment by jjcoffee** [comment]
> remove condition isLoading in Line 169 if not effect another PR

@charles-liang Generally it's best to know in advance if the solution would cause a regression. In this case it seems highly likely that this would stop the loader from showing in cases where we want it to show.

**Comment by jjcoffee** [comment]
Just a heads up I'll be OOO 6-13th, so feel free to re-assign if proposals start rolling in before I'm back!

**Comment by sakluger** [comment]
Reassigning C+ since JJ is out until next week.

**Comment by sakluger** [comment]
@thesahindia FYI we have an updated proposal from @charles-liang [here](https://github.com/Expensify/App/issues/40937#issuecomment-2096224073).

**Comment by tsa321** [comment]
## Proposal

Workspace expense chat becomes blank page when go back from IOU / transaction thread.

### What is the root cause of that problem?

In here:

https://github.com/Expensify/App/blob/2a6d63d7f17b1e4fba94f26103f13c6afe294b91/src/pages/home/report/ReportActionsView.tsx#L403-L415

The shouldOpenReport condition always become true for this issue because `hasCreatedAction` always return false.

https://github.com/Expensify/App/blob/2a6d63d7f17b1e4fba94f26103f13c6afe294b91/src/pages/home/report/ReportActionsView.tsx#L233-L234

This is because type of oldestReportAction is `INVITETOROOM` and the `CREATED` is secondLast action.

This will call the Report.openReport and will set ReportMetadata.isLoadingInitialReportActions to be true: in [here](https://github.com/Expensify/App/blob/2a6d63d7f17b1e4fba94f26103f13c6afe294b91/src/libs/actions/Report.ts#L747)
The isLoadingInitialReportActions value of true will cause isLoading to be true and will cause reportActions to be empty array. Set in here:

https://github.com/Expensify/App/blob/2a6d63d7f17b1e4fba94f26103f13c6afe294b91/src/pages/home/report/ReportActionsView.tsx#L165-L167

and So the report screen will be empty.


### What changes do you think we should make in order to solve the problem?

https://github.com/Expensify/App/blob/2a6d63d7f17b1e4fba94f26103f13c6afe294b91/src/pages/home/report/ReportActionsView.tsx#L233-L234

in here we must use last and second last action to check whether it is created action. For assurance we could check on the last 3 report action whether the report actions has CREATED type.

**Comment by jjcoffee** [comment]
@sakluger Happy to take this one back if @thesahindia is too busy!

**Comment by sakluger** [comment]
Thanks @jjcoffee! I've assigned it back to you.

**Comment by jjcoffee** [comment]
Actually not able to reproduce myself anymore. Are you guys able to repro still? @tsa321 @charles-liang @beodw

**Comment by tsa321** [comment]
@jjcoffee  I cannot reproduce anymore. 
Looks like already fixed from BE. The oldest `reportAction.actionName` is 'CREATED' and second oldest is `INVITETOROOM` :

![Screenshot 2024-05-16 at 05 48 43](https://github.com/Expensify/App/assets/114975543/4a31ffb3-0ef2-49c0-be6b-8f9d5901a737)

**Comment by jjcoffee** [comment]
@sakluger Can we ask applause to retest here just to make sure?

**Comment by sakluger** [comment]
Sure! I added the retest-weekly label.

I also reassigned to @dylanexpensify since I'll be OOO until May 31.

**Comment by jjcoffee** [comment]
Waiting for a retest.

**Comment by dylanexpensify** [comment]
Same ^

**Comment by jjcoffee** [comment]
Still waiting for a retest.

**Comment by dylanexpensify** [comment]
@lanitochka17 bump! 🙇‍♂️

**Comment by lanitochka17** [comment]
Issue is still reproducible on the latest build 1.4.78-3


https://github.com/Expensify/App/assets/78819774/522a5126-4ed0-47a3-993c-394e1315ee1a

**Comment by sakluger** [comment]
@jjcoffee, it looks like this is still reproducible. Are you able to reproduce it as well by following the steps in the latest reproduction video?

**Comment by tsa321** [comment]
## Proposal

@jjcoffee  [Updated](https://github.com/Expensify/App/issues/40937#issuecomment-2106180673)

**Comment by sakluger** [comment]
Oh yeah, I removed @dylanexpensify's assignment since I'm back.

**Comment by tsa321** [comment]
### Proposal

[Updated](https://github.com/Expensify/App/issues/40937#issuecomment-2106180673).  @jjcoffee I have updated my proposal and will add the video later

**Comment by tsa321** [comment]
@jjcoffee Here is video result using my updated proposal:

<details><summary>Video Result </summary>

https://github.com/Expensify/App/assets/114975543/dbe66d16-b13d-437f-ba19-bc5c407874cd

</details>


<details><summary>Before applying the fix </summary>

https://github.com/Expensify/App/assets/114975543/fb221f6b-1a28-4625-884e-d13f2338f99f

</details>

**Comment by jjcoffee** [comment]
@tsa321 The rest of the messages in the chat don't seem to load when I apply your main solution

`const isLoading = !!reportActionID && isLoadingInitialReportActions && !isReadyForCommentLinking;`

https://github.com/Expensify/App/assets/27287420/55317f9e-5ee0-4296-9413-8e434408e536

**Comment by jjcoffee** [comment]
@sakluger What do you think about bumping the price here to encourage deeper investigation? I think it's turning out to be quite complex!

**Comment by tsa321** [comment]
@jjcoffee I have found the root cause,:

When user submit expense report -> open expense report, the `OpenReport` Api request of expense report will wait for  `RequestMoney` Api Request to complete. But Onyx metadata `isLoadingInitialReportActions` for the report already set to true. The `RequestMoney` usually taking longer than `OpenReport` Api.
When user open transaction thread  then goback by click header link, if `RequestMoney` and `OpenReport` aren't completed yet, he will see blank expense report. This is because

https://github.com/Expensify/App/blob/bc3009a97dfb9eba4127e182c7703564166ba4c9/src/pages/home/report/ReportActionsView.tsx#L122

`isLoading` is true, because `isLoadingInitialReportActions` still true until `OpenReport` completed and `reportActionID` exist because user go back to expense report by clicking header from transaction thread. 

<br />
You can simulate the bug without submitting expense report, by fastly opening expense report -> open transaction thread by clicking on report action preview -> go back to expense report by clicking link in the header. Or you can try it by slowing down network.

**Comment by jjcoffee** [comment]
> There is second change in my updated proposal

@tsa321 I have applied the second change but now once `OpenReport` completes, the report preview jumps to the bottom and all following messages/report previews are removed. I'm a bit concerned that we're just trying different code changes here without understanding how they work/what they do.

> When user submit expense report -> open expense report, the OpenReport Api request of expense report will wait for RequestMoney Api Request to complete.

I don't think this RCA is correct, as if you submit a new expense after already loading the comment-linked report (i.e. with the URL pointing to a report action, e.g. `/r/4901005732703642/4149801980005824626`) this issue doesn't exist (the report and all actions load immediately).

**Comment by mallenexpensify** [comment]
@jjcoffee for potentially increasing the price here, can you follow the process [here](https://docs.google.com/document/d/1BvohU05MTaHnjOD_vwJv_aqDAirv-ChkyRnKCAvOVyQ/edit#bookmark=id.25bt9k50o02o).  Can you also tag me in the #contributor-plus post too. Thx

**Comment by jjcoffee** [comment]
@mallenexpensify Is that the right process for bumping the price to encourage higher quality proposals? It seems to be more for more unclear situations. Maybe I'm missing something though!

**Comment by mallenexpensify** [comment]
> What do you think about bumping the price here to encourage deeper investigation? 

@jjcoffee I def had that wrong (so wrong I wonder if I posted on the wrong issue)
I updated the price to $500.

**Comment by jjcoffee** [comment]
@mallenexpensify Thanks and no worries, it happens! :smile: I've bumped for more proposals on Slack now.

@tsa321 Could you update your proposal with the latest findings, so it's easier to review it as a whole? I'm looking for a clear explanation of why the proposed changes work to fix the RCA, rather than just that they work. The comment linking functionality is quite complex so we have to be quite careful about regressions.

I'd also like to address why this only seems to happen if you're not already comment linked to the IOU report action as I mentioned [here](https://github.com/Expensify/App/issues/40937#issuecomment-2152560567).

**Comment by tsa321** [comment]
@jjcoffee I have update my [proposal](https://github.com/Expensify/App/issues/40937#issuecomment-2106180673)
and I have rephrased my RCA. Maybe there is misunderstanding.

> I don't think this RCA is correct, as if you submit a new expense after already loading the comment-linked report (i.e. with the URL pointing to a report action, e.g. /r/4901005732703642/4149801980005824626) this issue doesn't exist (the report and all actions load immediately).

I have rephrased my RCA, so more details in there. 
But basically: let's say there are three reports:
1. workspace expense report (this report has a report action expense preview with text for example `8 expenses`)
2. report which contains many report action previews of expense requests
3. the transaction details thread report from the report action preview of an expense in second report.

The issue appear if user submit expense from first report. The flow is: 
1. user submit expense from first report (will trigger MoneyRequest Api request)-> 
2. user open second report by clicking the expense preview report action (preview that contains text for example `8 expenses`) this will trigger OpenReport api request for second report -> 
3. user click an expense preview from second report which will open third report -> 
4. user goback to second report by clicking header link (this goback here will not trigger `OpenReport` for second report because there is report action in route/url) .

The problem here is because the `OpenReport` from second step isn't completed yet. This is because the open report is waiting for `MoneyRequest` to complete, but `isLoadingInitialReportActions` already set to true for the second report. 
 
<br />
 
The comment linking feature is introduced recently: https://github.com/Expensify/App/pull/30269

It is relatively new, and I think the additional problems are related to `isLoadingInitialReportActions` set to true without checking on the cached messages.

**Comment by jjcoffee** [comment]
@tsa321 Thanks for the clarification, but I think there's still a lack of explanation of what `(!!reportActionID && isLoadingInitialReportActions)` does, which makes it hard for me to evaluate whether or not changing that condition is safe against regressions. Can you clarify what it's purpose is? I imagine there's a specific case it was added to handle, so it would be good to check against it.

In general it seems like we're having a lot of problems with `isLoadingInitialReportActions` (considering your [other](https://github.com/Expensify/App/issues/40937#issuecomment-2158931087) proposed change), and I can see that if we change `isLoadingInitialReportActions` to false in `OpenReport` it "fixes" the issue. So it might be worth looking into what it's actually meant to be used for and if we need it in this case. I assume that removing it entirely from the dependencies would cause regressions, as it's probably pretty key for pagination.

**Comment by tsa321** [comment]
> here's still a lack of explanation of what (!!reportActionID && isLoadingInitialReportActions)

I am not sure too. this value will determine whether the reportActions should be empty array when OpenReport executed and there is reportActionID. 
This check also exist in `ReportScreen` if these condition meet the report screen will diaplays `ReportActionsSkeletonView`. So the logic already handled in `ReportScreen`. 
But in `ReportScreen` the condition also consider report actions cache, if linked report action cache available the `ReportActionsSkeletonView` won't be displayed. 

<br />
and in here:

https://github.com/Expensify/App/blob/15a6ef4121c12bd086441a1a8b39fb0fdf54e96e/src/pages/home/report/ReportActionsView.tsx#L144-L155

I think the logic of isLoading and the changes of the key of FlatList by listID which both variables depend on `isLoadingInitialReportActions` is : `Change the list ID only for comment linking to get the positioning right`. 
But I have tested my proposed changes the position of linked report action is correct.

**Comment by sakluger** [comment]
Latest update: we're still working on getting the proposal right.

**Comment by jjcoffee** [comment]
Asking for clarification on `isLoadingInitialReportActions` from the comment linking PR author [here](https://github.com/Expensify/App/pull/30269/files#r1642624582).

**Comment by sakluger** [comment]
No response - I bumped the comment. @jjcoffee there are a lot of comments on that PR, it may help to raise in Slack.

**Comment by sakluger** [comment]
Looks like we got a response to that comment.

**Comment by jjcoffee** [comment]
@tsa321 Are you able to test your changes using the test steps in https://github.com/Expensify/App/pull/30269? That should cover my remaining concerns, I think!

**Comment by tsa321** [comment]
@jjcoffee 


<details><summary>Test 1 & 2:</summary>

https://github.com/Expensify/App/assets/114975543/dd7f90af-64dc-4f2b-ada7-a324755a076e

</details>


<details><summary>Test 3:</summary>


https://github.com/Expensify/App/assets/114975543/55be631a-9598-46b8-8921-ef560672b04f

</details>

**Comment by jjcoffee** [comment]
Thanks for your hard work on this @tsa321 - I'm convinced we can go with your [proposal](https://github.com/Expensify/App/issues/40937#issuecomment-2106180673) now!

:ribbon::eyes::ribbon: C+ reviewed

**Comment by marcaaron** [comment]
@roryabraham Could we get someone from Margelo to look at this and validate the solution? It seems possibly related to comment linking. There's some discussion of a "cached linked report action" and `isReadyForCommentLinking` param, but I'm not confident in my understanding of how this works to approve this solution and don't feel like I have a solid grasp on what is being proposed.

@jjcoffee what's your take on it? The conditions in the report page are starting to look a bit complex here. Can you summarize it?

**Comment by marcaaron** [comment]
>The core of it is that we are setting isLoading to true when we already have all the information necessary to display the report actions. Because there is an OpenReport request queued after the RequestMoney request (which is quite slow) this leads to quite a delay before we display anything.

`RequestMoney` sets an `isLoading` and that somehow prevents the report page from loading with an initial set of stored actions? That can't be right, can it?

When and where is the `OpenReport` call triggered?

I'm not sure, but it seems like the complexity of this code is a bit high. I'm nervous about making layering more hacks onto the existing complexity. Is there a step by step explanation of how the bug happens? Maybe linking to specific examples in the code.

**Comment by marcaaron** [comment]
I generally understood this explanation to mean that the `RequestMoney` call is somehow blocking `OpenReport` from completing and leading to the blank page:

>When user open expense report (let's say this is first report)-> submit expense -> click on report action preview of expenses (report action expense preview which contains words like 5 expenses) -> a report which contains list of report action expense previews will be opened (let's say this is second report) -> click one of the expense report action preview (which will open transaction detail thread (let's say this is third report)) -> go back to the second report by clicking on header link: the OpenReport Api request of the second report (the OpenReport API triggered by clicking on expense preview from first report) will wait for RequestMoney Api Request to complete. But Onyx metadata isLoadingInitialReportActions for the second report already set to true. The RequestMoney API request usually taking longer to complete than OpenReport, but OpenReport here must wait for RequestMoney to complete.

But it's very difficult to read. Can we break down how the bug happens step by step with links to the code please and then update the proposal?

**Comment by tsa321** [comment]
@marcaaron yes, please wait I will update the explanation.

**Comment by tsa321** [comment]
@marcaaron, I have updated my [proposal](https://github.com/Expensify/App/issues/40937#issuecomment-2106180673) to make it easier to read.

**Comment by sakluger** [comment]
@marcaaron what do you think of the updated proposal?

**Comment by marcaaron** [comment]
Ok, I understand that `OpenReport` is called when we tap the expense preview.

>The problem arises because the OpenReport process initiated in the second step has not yet completed. This delay occurs because the OpenReport is waiting for the MoneyRequest to finish, while isLoadingInitialReportActions is set to true for the second report

But we are loading report actions for which report exactly? What is the "second" report? Could you possibly update your answer to be more specific?

>Consequently, isLoading remains true due to the presence of a reportActionID in the route

What is the function of the `reportActionID`? And why do we set `isLoading` true when we have one?

**Comment by jjcoffee** [comment]
Here's my take on the RCA:

1. Submitting an expense triggers a `RequestMoney` request
2. Tapping the header link **comment-links** you to the `reportAction` for the expense in the WS chat, so we have `reportActionID` in the route (which is what causes this to be a comment link)
3. This link calls `OpenReport` which adds a request to the queue and sets `isLoadingInitialReportActions` to true
4. The `RequestMoney` request is (very) slow, so the `OpenReport` request is just waiting in the request queue and will generally not be completed once you're navigated in (2)
5. `isLoadingInitialReportActions` remains true whilst the `OpenReport` request is pending, so the `isLoading` here is true (as we also have `reportActionID` in the route):

https://github.com/Expensify/App/blob/bc3009a97dfb9eba4127e182c7703564166ba4c9/src/pages/home/report/ReportActionsView.tsx#L122

6. Since `isLoading` is true, the `reportActions` array is set to empty here:
https://github.com/Expensify/App/blob/2a6d63d7f17b1e4fba94f26103f13c6afe294b91/src/pages/home/report/ReportActionsView.tsx#L165-L167
7. The conditions for `shouldShowSkeleton` do not match the conditions for `isLoading`, so we just display a blank screen.
https://github.com/Expensify/App/blob/9fbbbfb01a64c8bc7214fc6d04ab71e61d15bda6/src/pages/home/ReportScreen.tsx#L410-L416

Side-note - `shouldShowSkeleton` is actually passed via `isReadyForCommentLinking` here:

https://github.com/Expensify/App/blob/9fbbbfb01a64c8bc7214fc6d04ab71e61d15bda6/src/pages/home/ReportScreen.tsx#L784

**Comment by jjcoffee** [comment]
> What is the function of the `reportActionID`? And why do we set `isLoading` true when we have one?

Having `reportActionID` indicates that we're comment-linking, so we need to build the `reportActions` array for that purpose. Looks like `isLoading` was originally added [here](https://github.com/Expensify/App/commit/d08dea532480085710e2acc190401eca7af9ff1c) and there's a clue in the commit message - "fix issue with navigating between different reports when actions are cached", but it's not clear why there needs to be a separate condition from just `isReadyForCommentLinking`. My guess is that `isLoadingInitialReportActions` is being used as a proxy for "we haven't got any reportActions yet", but I'm not sure about it. Would be great to get @perunt to shed some light on that!

**Comment by jjcoffee** [comment]
@tsa321 I see that you're working on https://github.com/Expensify/App/issues/43791 which is related. Does it impact this issue at all? I see that you've proposed to adjust `shouldShowSkeleton` and wonder if we should hold on that issue.

**Comment by tsa321** [comment]
@jjcoffee yes, I think it is better to hold for that issue.

**Comment by marcaaron** [comment]
>This link calls OpenReport which adds a request to the queue and sets isLoadingInitialReportActions to true

Is it an expected behavior to block on the `OpenReport` call in this case? At a high level - it seems like that might be the intended behavior if we are setting a variable called `isLoadingInitialReportActions` to `true`.

The expected behavior in the issue description says:

>Collect WS chat with history opens

So, what is the expected behavior? To open with "history" or to show a loading state?

**Comment by marcaaron** [comment]
also someone please provide more context about why we should put this issue on HOLD, thanks!

**Comment by marcaaron** [comment]
>The conditions for shouldShowSkeleton do not match the conditions for isLoading, so we just display a blank screen.

Sorry, by "match the conditions" do you mean to say `isLoading` is `true` when `shouldShowSkeleton` is `false`?

Why do we need both `isLoading` and `shouldShowSkeleton`? Are they not the same thing?

**Comment by jjcoffee** [comment]
> Is it an expected behavior to block on the OpenReport call in this case? At a high level - it seems like that might be the intended behavior if we are setting a variable called isLoadingInitialReportActions to true.

I don't think so, since we don't do it when we are not comment linking (i.e. if you just open the report without a comment link it loads immediately even though there is also an `OpenReport` call). I can't think of any reason why we'd need to do it specifically for comment linking either.

> what is the expected behavior? To open with "history" or to show a loading state?

I've assumed the expected behaviour is to open with the cached report actions, as there's only a need to display a loader if we actually have nothing at all to show (e.g. a fresh sign-in).

> please provide more context about why we should put this issue on HOLD, thanks!

@tsa321 is working on https://github.com/Expensify/App/issues/43791 which impacts this issue as it will adjust the conditions for `shouldShowSkeleton`. (That issue is also held on https://github.com/Expensify/App/issues/44625, where comment linking scrolling isn't working correctly, so might also make testing for regressions here a bit tricky).

> Sorry, by "match the conditions" do you mean to say isLoading is true when shouldShowSkeleton is false?

Yes, exactly. So a simple (workaround) solution would be just to change the conditions so that we show the skeleton, but I don't think that's really expected here.

> Why do we need both isLoading and shouldShowSkeleton? Are they not the same thing?

`shouldShowSkeleton` is specific to `ReportScreen` and isn't reliant on whether or not we are comment linking. `isLoading` is used in `ReportActionsView` specifically for comment linking/pagination. As for why we don't just use `shouldShowSkeleton` in place of `isLoading`, I don't think we have a clear answer on that. I did test doing that a while ago and it sort of works, but then the linked report action jumps around on the screen. Maybe @tsa321 can chime in here, or @perunt if he's around.

**Comment by marcaaron** [comment]
>I don't think so, since we don't do it when we are not comment linking

Let's confirm this first and then proceed. We _are_ linking to a particular comment in this case, right?

**Comment by marcaaron** [comment]
@roryabraham I'm assigning you here to help whenever you can. We can't seem to agree on a solution and this looks somehow related to comment linking.

**Comment by jjcoffee** [comment]
> We are linking to a particular comment in this case, right?

Yes, we're linking to the IOU action in the workspace chat. If we changed the header behaviour to remove the comment link (i.e. just link to the workspace chat only), we'd not see this issue.

**Comment by jjcoffee** [comment]
@sakluger Can we add a hold for https://github.com/Expensify/App/issues/43791 here and switch to weekly? The solution there affects the one here so I think it's best we revisit this once that one is sorted, unless @marcaaron feels strongly otherwise of course!

**Comment by sakluger** [comment]
Still on hold.

**Comment by marcaaron** [comment]
@jjcoffee are we ready to get this one moving again?

**Comment by jjcoffee** [comment]
@marcaaron Yes, looks like https://github.com/Expensify/App/issues/43791 is merged so we're ready to move forward here! Was there anything else you needed to clarify with @tsa321's [proposal](https://github.com/Expensify/App/issues/40937#issuecomment-2106180673)?

**Comment by marcaaron** [comment]
It's been a while but I've been waiting on an answer this:

>shouldShowSkeleton is specific to ReportScreen and isn't reliant on whether or not we are comment linking. isLoading is used in ReportActionsView specifically for comment linking/pagination. As for why we don't just use shouldShowSkeleton in place of isLoading, I don't think we have a clear answer on that. I did test doing that a while ago and it sort of works, but then the linked report action jumps around on the screen. Maybe @tsa321 can chime in here, or @perunt if he's around.

@tsa321 can you chime in / are you still interested in working on this issue?

**Comment by marcaaron** [comment]
Ok, let's give it a shot and test.

**Comment by jjcoffee** [comment]
PR is under review, just one little issue that I think is also there on main, but otherwise testing really well!

**Comment by jjcoffee** [comment]
I discovered [an issue](https://github.com/Expensify/App/pull/46864#issuecomment-2273743625) on native (iOS & Android) whilst testing the PR that was also there on main. We decided to fix it as a follow-up and @tsa321 has a solution, so @sakluger would you be able to create an issue and assign us both? (I'm not sure what the process is otherwise!).

Repro steps (video [here](https://github.com/Expensify/App/pull/46864#issuecomment-2273743625)):

1. From the main workspace chat, submit an expense request using the + button on the left side of the composer.
2. Complete the expense request submission process.
3. Click on the expense preview in the main workspace chat to open the report with all expense previews.
4. Click on the 3rd expense preview from the last (bottom) expense message/last expense preview. This will open the transaction thread report.
5. Immediately go back by clicking on the header link.
6. Try to scroll down

**Comment by jjcoffee** [comment]
[Deployed](https://github.com/Expensify/App/pull/46864#issuecomment-2289860684) to production 13 Aug, @sakluger can you update the title to hold for payment for 20 Aug? :pray:

**Comment by jjcoffee** [comment]
- The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/30269
- The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/30269/files#r1642624582
- A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A - comment linking was a complex PR so it makes sense some issues were missed
- Determine if we should create a regression test for this bug. **Yes**
- If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

**Regression Test Proposal**

1. If there aren't already, add some expenses to any workspace chat, e.g. 5.
2. Open each expense to view the transaction details report.
3. Go back to the workspace chat and submit a new expense using the + button on the left side of the composer.
4. Click on the expense preview in the chat to open the report with all expense previews.
5. Click on the 3rd expense from the bottom. This will open the transaction thread report.
6. Immediately go back by clicking on the header link.
7. Verify that there is no loading skeleton in the report and that the message loads immediately.

Do we agree 👍 or 👎

**Comment by jjcoffee** [comment]
@sakluger Payment is due today, as this was [deployed](https://github.com/Expensify/App/pull/46864#issuecomment-2289860684) 13 Aug.

**Comment by jjcoffee** [comment]
@marcaaron Are you able to create the follow-up issue we discussed? Details [here](https://github.com/Expensify/App/issues/40937#issuecomment-2284218714). Or is there some other process we should follow?

**Comment by marcaaron** [comment]
yes, report bug in #expensify-bugs

**Comment by jjcoffee** [comment]
@sakluger Friendly bump for payment (the fix was [deployed](https://github.com/Expensify/App/pull/46864#issuecomment-2289860684) 13 Aug) :pray:

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor: @tsa321 $500, paid via Upwork
Contributor+: @jjcoffee $500, paid via Upwork

**Comment by sakluger** [comment]
@jjcoffee thanks for being on top of everything here! And sorry for the delayed payment. All done now 🙇

**Comment by jjcoffee** [comment]
@sakluger Thanks for making the payment! It looks like the job price didn't update to $500, any way you can correct it?

![image](https://github.com/user-attachments/assets/e11993fa-0185-42c7-ba8e-b47a07545d60)

**Comment by sakluger** [comment]
@jjcoffee - sorry about that! I added a $250 bonus to the contract to bring the total payment to $500.
